synop2bufr/cli.py

```diff
@@ -98,17 +98,66 @@ def data():
 def transform(ctx, synop_file, metadata, output_dir, year, month, verbosity):
 
     try:
-        result = transform_synop(synop_file.read(), metadata.read(),
-                                 year, month)
+        # Get content from synop file
+        content = synop_file.read()
+
+        # Split multiple messages by GTS end of message signal NNNN
+        messages = content.upper().split("NNNN")
+
+        # Remove leading or trailing whitespaces from these messages
+        # and ignore empty messages after the final NNNN
+        messages = [msg.strip() for msg in messages if msg != ""]
+
+        # Read metadata file contents as a string
+        metadata_string = metadata.read()
+
+        # Transform each message to BUFR files
+        for synop_msg in messages:
```

Comment:
Sorry, this has just occurred to me as part of writing the pygeoapi plugin. When data are uploaded to the WIS2box the cli will not be called. Do we need to move this split into the transform function instead?